src/main/java/org/squiddev/plethora/integration/vanilla/DisableAI.java

```diff
@@ -104,4 +105,75 @@ public void deserializeNBT(NBTTagCompound tag) {
 			disabled = tag != null && tag.getBoolean("disabled");
 		}
 	}
+
+	public static class AIPlethoraObedience extends EntityAIBase {
+
+		private EntityLiving entity;
+		private EntityLookHelper oldInstance;
+
+		AIPlethoraObedience(EntityLiving entityIn) {
+			this.entity = entityIn;
+		}
+
+		private boolean isAIDisabled() {
+			DisableAI.IDisableAIHandler disable = this.entity.getCapability(DisableAI.DISABLE_AI_CAPABILITY, null);
+			if (disable == null) return false;
+			return disable.isDisabled();
+		}
+
+		@Override
+		public void startExecuting() {
+			// This should not be already set when we start executing because it should have been reset.
+			assert (!( this.entity.getLookHelper() instanceof PlethoraEntityLookHelper));
+			this.oldInstance = this.entity.getLookHelper();
+			ReflectionHelper.setPrivateValue(EntityLiving.class, this.entity, new PlethoraEntityLookHelper(), "lookHelper");
```

Comment:
From memory, `lookHelper` is final - slightly surprised this works :). Overwriting this feels _really_ icky, though I'm really not sure if any of the other proposed solutions are any better.

Anyway, just a couple of points:
 - Might be worth using [`TypedField`][tf] instead of `ReflectionHelper`. It caches the field (and provides a little better safety), which makes it a little more efficient.
 - You'll need to use both the obfuscated field (`field_70749_g`) and deobfuscated one.
 - One thing we could consider, but maybe not worth it, is getting `PlethoraEntityLookHelper` to have an enabled/disabled field, and just toggle that after injecting it the first time.

[tf]: https://github.com/SquidDev-CC/plethora/blob/minecraft-1.12/src/main/java/org/squiddev/plethora/utils/TypedField.java

Reply:
Regarding point two, ReflectionHelper does the lookup for us, where as TypedField does not. Is it worth taking a lesson from ReflectionHelper and only using the deobf'd name and lookup the obf name?

Reply:
Regarding point one, since this is only set and unset once when the AI is enabled or disabled, I'm not sure how its more efficient.

Point three, I'm happy with that approach, however I think its better we leave no trace we were there when functionality is returned to normal. We can't even be sure another AITask doesn't expect *their* own implementation of LookHelper, which is really why we should have a Forge hook for this, but I'm not feeling up to designing, pulling, reviewing a forge change.

Reply:
Yeah, not sure why I'm not using `FMLDeobfuscatingRemapper.INSTANCE`. I know I looked into it at the time, but haven't the foggiest why I didn't end up using it.

Shouldn't be too hard to add that, I'll have a fiddle.

Reply:
That said, you do have to provide the obf/MCP name, rather than the deobf one (as obf won't change, but deobf will). I think that's part of why I have the "must specify both" thing - makes it super clear what field you are actually using.

Reply:
Is the deobf name more likely to change than the obf name once it has been given a sensible name? Its my understanding that obf names will change with every minecraft build, where as deobf names should be consistent for future versions of Forge. Is this assumption incorrect?

Reply:
From discussion in Discord, we will go with the SRG names as they change least often.